src/MMCPServer.cpp

```diff
@@ -0,0 +1,988 @@
+            groupNotEmpty = true;
+            cl->writeData(outMsg);
+        }
+    }
+
+    using namespace AnsiColors;
+    /*: %1 and %3 are ASCII ESC color codes that need to be included BEFORE a
+     * portion of text (the main message %5) and (the group name %4)
+     * respectively and %6 is another code at the very end to reset the colors
+     * back to "normal". %2 is the prefix added to all chat messages display to us.
+     * Please try and reproduce the positioning of those codes around the translation.
+     */
+    if (groupNotEmpty) {
+        clientMessage(tr("%1%2You chat to %3<%4>%1, '%5'%6")
+                              .arg(FBLDRED, mpHost->getMMCPChatPrefix(), FBLDCYN, group, message, RST));
+        return {true, QString()};
+    }
+
+    /*: %1 and %3 are ASCII ESC color codes that need to be included BEFORE a
+     * portion of text (the main message %5) and (the group name %4)
+     * respectively and %5 is another code at the very end to reset the colors
+     * back to "normal". %2 is the prefix added to all chat messages display to us.
+     * Please try and reproduce the positioning of those codes around the translation.
+     */
+    clientMessage(tr("%1%2You try to chat to <%3%4%1> but it is empty and no-one hears you say: '%5'%6")
+                          .arg(FBLDRED, mpHost->getMMCPChatPrefix(), FBLDCYN, group, message, RST));
+    return {false, qsl("nobody in group '%1' now").arg(group)};
+}
+
+/**
+ * Script command, Display a list of connected chat clients on the main console
+ */
+QPair<bool, QString> MMCPServer::chatList()
+{
+    using namespace AnsiColors;
+
+    QStringList peersList;
+    QListIterator<QPointer<MMCPClient>> it(mPeersList);
+    while (it.hasNext()) {
+        MMCPClient* pClient = it.next();
+       
+        peersList << qsl("%1%2 %3%4 %5")
+                            .arg(pClient->isPending() ? FBLDYEL : FBLDGRN)
+                            .arg(pClient->id(), 4)
+                            .arg(RST, pClient->getInfoString(), pClient->getVersion());
+    }
+
+    const QString strMessage = tr("%1Id   Name                 Address              Port  Group           Flags    ChatClient\n"
+                                  "==== ==================== ==================== ===== =============== ======== ================\n"
+                                  "%2"
+                                  "==== ==================== ==================== ===== =============== ======== ================\n"
+                                  "Color Key: %3Connected  %4Pending%1\n"
+                                  "Flags:  A - Allow Commands, F - Firewall, I - Ignore,  P - Private   n - Allow Snooping\n"
+                                  "        N - Being Snooped,  S - Serving,  T - Allows File Transfers, X - Serve Exclude%1")
+                                       .arg(RST,
+                                            peersList.isEmpty() ? QString() : peersList.join(QChar::LineFeed),
```

Comment:
```suggestion
                                            peersList.isEmpty() ? QString() : qsl("%1\n").arg(peersList.join(QChar::LineFeed)),
``` last item in list has the footer on same line